src/edu/ucsb/cs56/projects/games/fish_animation/FishAnimationEnvironment.java

```diff
@@ -578,22 +579,29 @@ public void display(int delay, DrawingPanel fishPanel)
 				}
 
 				// Speed of the fish moving across the screen
-				double currentSpeed = numFish % 6;
+				double currentSpeed = numFish % 2;
```

Comment:
I see you've made a bunch of changes in this section.  Can you explain what they're doing?  I can't tell just by looking through the code.

Reply:
For this one, I change 6 to 2 to make the fish swim more fluently.

Reply:
I get that you changed 6 to 2 - that's obvious.  My question is about your new algorithm.  How does it work?  How does it make the fist swim more smoothly?

Reply:
% 2 results in 0, 1 and 2 which are in a smaller range of speed.

Reply:
I'm talking about the whole algorithm, here.  Lines ~580 through ~605.  In particular:
 - What does using 2 instead of 6 accomplish?  Doesn't it just get overwritten a few lines later?
 - The part with the random numbers looks weird and complicated.  What is it trying to do?  How does it do it?
 - That `if` statement at the bottom.  What does it do?  How does it do it?

Reply:
The random part gives a random Y position of the fish and if statement judge whether the fish is out of the upper or lower bound of the screen. The original version of the code only change X position so it does not have a if condition for Y position. 
To be honest, 2 or 6 actually does not matter a lot. 2 gives a more fluent move since 2 gives a small range of speed (0 - 1).

Reply:
Okay - that makes sense.  I'm still not convinced it's ever used, though.  Look at the condition on line 584.  Doesn't `currentSpeed` get immediately reset?

I'm pretty sure the random parts can be simplified down to a single call to `nextInt()`.  You should take a look at that.

And the `if` statement only handles fish that go off one of the edges (the top, I think).  What about the ones that go off the bottom?

Finally - and this isn't part of your PR - but I only see fish getting _added_ to `fishArray`.  Are they ever removed?  This might be part of the game's performance problems...

Reply:
ummm...I didn't notice that part. Let me change it back to 6.

For the random, I want the range of speed to be (1,3) or (-3, -1) so I used two nextInt(). I'm not sure whether a single nextInt() could give me a range of (1,3) or (-3, -1).

I just fix the newY if statement to handle fish that go off the bottom.

I actually don't know how to delete the fish:)

I heard from Sitao that you live in BreakPointe. Maybe I can show you my code tomorrow if you have time:)